<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</title>
  <meta name="description" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018" />
  
  
  

<meta name="author" content="Sudarshan A. Shetty, Leo Lahti, Gerben DA. Hermes">


<meta name="date" content="2018-05-29">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="linear-models-the-role-of-covariates.html">
<link rel="next" href="multivariate-comparisons-of-microbial-community-composition.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.4/datatables.js"></script>
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#preparing-for-the-course"><i class="fa fa-check"></i><b>1.1</b> Preparing for the course</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#useful-functionsresources"><i class="fa fa-check"></i><b>1.2</b> Useful functions/resources</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#focus"><i class="fa fa-check"></i><b>1.3</b> Focus</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#target-audience"><i class="fa fa-check"></i><b>1.4</b> Target audience</a></li>
<li class="chapter" data-level="1.5" data-path="introduction.html"><a href="introduction.html#license"><i class="fa fa-check"></i><b>1.5</b> License</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html"><i class="fa fa-check"></i><b>2</b> Set-up and Pre-processing</a><ul>
<li class="chapter" data-level="2.1" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-or-asvs-or-sotus"><i class="fa fa-check"></i><b>2.1</b> OTU or ASVs or sOTUs</a></li>
<li class="chapter" data-level="2.2" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#general-overview"><i class="fa fa-check"></i><b>2.2</b> General overview</a></li>
<li class="chapter" data-level="2.3" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#structure"><i class="fa fa-check"></i><b>2.3</b> Structure</a></li>
<li class="chapter" data-level="2.4" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#making-a-phyloseq-object"><i class="fa fa-check"></i><b>2.4</b> Making a phyloseq object</a></li>
<li class="chapter" data-level="2.5" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-input-to-phyloseq-object"><i class="fa fa-check"></i><b>2.5</b> Read input to phyloseq object</a></li>
<li class="chapter" data-level="2.6" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-the-tree-file."><i class="fa fa-check"></i><b>2.6</b> Read the tree file.</a></li>
<li class="chapter" data-level="2.7" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#merge-into-phyloseq-object."><i class="fa fa-check"></i><b>2.7</b> Merge into phyloseq object.</a></li>
<li class="chapter" data-level="2.8" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#read-data-from-otu-picking-stratergy"><i class="fa fa-check"></i><b>2.8</b> Read data from OTU-picking stratergy</a></li>
<li class="chapter" data-level="2.9" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#variablity"><i class="fa fa-check"></i><b>2.9</b> Variablity</a></li>
<li class="chapter" data-level="2.10" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#otu-picking-stratergy"><i class="fa fa-check"></i><b>2.10</b> OTU-picking stratergy</a></li>
<li class="chapter" data-level="2.11" data-path="set-up-and-pre-processing.html"><a href="set-up-and-pre-processing.html#ng-tax-stratergy"><i class="fa fa-check"></i><b>2.11</b> NG-tax stratergy</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="alpha-diversities.html"><a href="alpha-diversities.html"><i class="fa fa-check"></i><b>3</b> Alpha diversities</a><ul>
<li class="chapter" data-level="3.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#equal-sample-sums"><i class="fa fa-check"></i><b>3.1</b> Equal sample sums</a></li>
<li class="chapter" data-level="3.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#diversities"><i class="fa fa-check"></i><b>3.2</b> Diversities</a><ul>
<li class="chapter" data-level="3.2.1" data-path="alpha-diversities.html"><a href="alpha-diversities.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>3.2.1</b> Non-phylogenetic diversities</a></li>
<li class="chapter" data-level="3.2.2" data-path="alpha-diversities.html"><a href="alpha-diversities.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>3.2.2</b> Phylogenetic diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="composition-plots.html"><a href="composition-plots.html"><i class="fa fa-check"></i><b>4</b> Composition plots</a><ul>
<li class="chapter" data-level="4.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-counts"><i class="fa fa-check"></i><b>4.1</b> Barplot counts</a></li>
<li class="chapter" data-level="4.2" data-path="composition-plots.html"><a href="composition-plots.html#barplot-relative-abundance"><i class="fa fa-check"></i><b>4.2</b> Barplot relative abundance</a><ul>
<li class="chapter" data-level="4.2.1" data-path="composition-plots.html"><a href="composition-plots.html#barplot-customize"><i class="fa fa-check"></i><b>4.2.1</b> Barplot customize</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html"><i class="fa fa-check"></i><b>5</b> Beta diversity metrics</a><ul>
<li class="chapter" data-level="5.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>5.1</b> Phylogenetic beta-diversity metrics</a><ul>
<li class="chapter" data-level="5.1.1" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#unweighted-unifrac"><i class="fa fa-check"></i><b>5.1.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="5.1.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#weighted-unifrac"><i class="fa fa-check"></i><b>5.1.2</b> Weighted Unifrac</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#population-level-density-landscapes"><i class="fa fa-check"></i><b>5.2</b> Population-level Density landscapes</a></li>
<li class="chapter" data-level="5.3" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#permanova"><i class="fa fa-check"></i><b>5.3</b> PERMANOVA</a></li>
<li class="chapter" data-level="5.4" data-path="beta-diversity-metrics.html"><a href="beta-diversity-metrics.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>5.4</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>6</b> Core microbiota</a><ul>
<li class="chapter" data-level="6.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-microbiota-anlaysis"><i class="fa fa-check"></i><b>6.1</b> Core microbiota anlaysis</a></li>
<li class="chapter" data-level="6.2" data-path="core-microbiota.html"><a href="core-microbiota.html#core-abundance-and-diversity"><i class="fa fa-check"></i><b>6.2</b> Core abundance and diversity</a></li>
<li class="chapter" data-level="6.3" data-path="core-microbiota.html"><a href="core-microbiota.html#core-visualization"><i class="fa fa-check"></i><b>6.3</b> Core visualization</a><ul>
<li class="chapter" data-level="6.3.1" data-path="core-microbiota.html"><a href="core-microbiota.html#core-heatmaps"><i class="fa fa-check"></i><b>6.3.1</b> Core heatmaps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html"><i class="fa fa-check"></i><b>7</b> Inference of Microbial Ecological Networks</a><ul>
<li class="chapter" data-level="7.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-spieceasi"><i class="fa fa-check"></i><b>7.1</b> Prepare data for SpiecEasi</a></li>
<li class="chapter" data-level="7.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#spiec-easi-network-reconstruction"><i class="fa fa-check"></i><b>7.2</b> SPIEC-EASI network reconstruction</a><ul>
<li class="chapter" data-level="7.2.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#prepare-data-for-plotting"><i class="fa fa-check"></i><b>7.2.1</b> Prepare data for plotting</a></li>
<li class="chapter" data-level="7.2.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#igraph-network"><i class="fa fa-check"></i><b>7.2.2</b> igraph network</a></li>
<li class="chapter" data-level="7.2.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot"><i class="fa fa-check"></i><b>7.2.3</b> Network plot</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-properties"><i class="fa fa-check"></i><b>7.3</b> Network properties</a><ul>
<li class="chapter" data-level="7.3.1" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#modularity-in-networks"><i class="fa fa-check"></i><b>7.3.1</b> Modularity in networks</a></li>
<li class="chapter" data-level="7.3.2" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#network-plot-1"><i class="fa fa-check"></i><b>7.3.2</b> Network plot</a></li>
<li class="chapter" data-level="7.3.3" data-path="inference-of-microbial-ecological-networks.html"><a href="inference-of-microbial-ecological-networks.html#good-reads-for-ecological-networks"><i class="fa fa-check"></i><b>7.3.3</b> Good reads for ecological networks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html"><i class="fa fa-check"></i><b>8</b> Differential abundance testing for univariate data</a><ul>
<li class="chapter" data-level="8.1" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#load-example-data"><i class="fa fa-check"></i><b>8.1</b> Load example data</a></li>
<li class="chapter" data-level="8.2" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#visual-comparison-of-two-groups"><i class="fa fa-check"></i><b>8.2</b> Visual comparison of two groups</a></li>
<li class="chapter" data-level="8.3" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#statistical-comparison-of-two-groups"><i class="fa fa-check"></i><b>8.3</b> Statistical comparison of two groups</a></li>
<li class="chapter" data-level="8.4" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#investigate-assumptions-of-the-t-test"><i class="fa fa-check"></i><b>8.4</b> Investigate assumptions of the t-test</a></li>
<li class="chapter" data-level="8.5" data-path="differential-abundance-testing-for-univariate-data.html"><a href="differential-abundance-testing-for-univariate-data.html#compare-results-between-parametric-and-non-parametric-tests"><i class="fa fa-check"></i><b>8.5</b> Compare results between parametric and non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html"><i class="fa fa-check"></i><b>9</b> Linear models: the role of covariates</a><ul>
<li class="chapter" data-level="9.1" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#fitting-a-linear-model"><i class="fa fa-check"></i><b>9.1</b> Fitting a linear model</a></li>
<li class="chapter" data-level="9.2" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#interpreting-linear-model-output"><i class="fa fa-check"></i><b>9.2</b> Interpreting linear model output</a></li>
<li class="chapter" data-level="9.3" data-path="linear-models-the-role-of-covariates.html"><a href="linear-models-the-role-of-covariates.html#covariate-testing"><i class="fa fa-check"></i><b>9.3</b> Covariate testing</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html"><i class="fa fa-check"></i><b>10</b> Advanced models for differential abundance</a><ul>
<li class="chapter" data-level="10.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#particular-properties-of-taxonomic-profiling-data"><i class="fa fa-check"></i><b>10.1</b> Particular properties of taxonomic profiling data</a><ul>
<li class="chapter" data-level="10.1.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#discrete-count-data"><i class="fa fa-check"></i><b>10.1.1</b> Discrete count data</a></li>
<li class="chapter" data-level="10.1.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#sparsity"><i class="fa fa-check"></i><b>10.1.2</b> Sparsity</a></li>
<li class="chapter" data-level="10.1.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#rarity"><i class="fa fa-check"></i><b>10.1.3</b> Rarity</a></li>
<li class="chapter" data-level="10.1.4" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#overdispersion"><i class="fa fa-check"></i><b>10.1.4</b> Overdispersion</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#generalized-linear-models-a-brief-overview"><i class="fa fa-check"></i><b>10.2</b> Generalized linear models: a brief overview</a></li>
<li class="chapter" data-level="10.3" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#deseq2-differential-abundance-testing-for-sequencing-data"><i class="fa fa-check"></i><b>10.3</b> DESeq2: differential abundance testing for sequencing data</a><ul>
<li class="chapter" data-level="10.3.1" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#fitting-deseq2"><i class="fa fa-check"></i><b>10.3.1</b> Fitting DESeq2</a></li>
<li class="chapter" data-level="10.3.2" data-path="advanced-models-for-differential-abundance.html"><a href="advanced-models-for-differential-abundance.html#comparison-between-deseq2-and-standard-models"><i class="fa fa-check"></i><b>10.3.2</b> Comparison between DESeq2 and standard models</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html"><i class="fa fa-check"></i><b>11</b> Multivariate comparisons of microbial community composition</a><ul>
<li class="chapter" data-level="11.1" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#permanova-1"><i class="fa fa-check"></i><b>11.1</b> PERMANOVA</a></li>
<li class="chapter" data-level="11.2" data-path="multivariate-comparisons-of-microbial-community-composition.html"><a href="multivariate-comparisons-of-microbial-community-composition.html#checking-the-homogeneity-condition-1"><i class="fa fa-check"></i><b>11.2</b> Checking the homogeneity condition</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="further-exercises.html"><a href="further-exercises.html"><i class="fa fa-check"></i><b>12</b> Further exercises</a><ul>
<li class="chapter" data-level="12.1" data-path="further-exercises.html"><a href="further-exercises.html#compositionality"><i class="fa fa-check"></i><b>12.1</b> Compositionality</a></li>
<li class="chapter" data-level="12.2" data-path="further-exercises.html"><a href="further-exercises.html#redundancy-analysis-rda"><i class="fa fa-check"></i><b>12.2</b> Redundancy analysis (RDA)</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">OPEN &amp; REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="advanced-models-for-differential-abundance" class="section level1">
<h1><span class="header-section-number">10</span> Advanced models for differential abundance</h1>
<p>GLMs are the basis for advanced testing of differential abundance in sequencing data. This is necessary, as the sequencing data sets deviate from symmetric, continuous, Gaussian assumptions in many ways.</p>
<div id="particular-properties-of-taxonomic-profiling-data" class="section level2">
<h2><span class="header-section-number">10.1</span> Particular properties of taxonomic profiling data</h2>
<div id="discrete-count-data" class="section level3">
<h3><span class="header-section-number">10.1.1</span> Discrete count data</h3>
<p>Sequencing data consists of discrete counts:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print</span>(<span class="kw">abundances</span>(d)[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>])</code></pre></div>
<pre><code>##                              Sample-1 Sample-2 Sample-3
## Actinomycetaceae                   11       67       21
## Aerococcus                          1        1        1
## Aeromonas                           1        1        1
## Akkermansia                      1167     6127     4235
## Alcaligenes faecalis et rel.       90      126      188</code></pre>
</div>
<div id="sparsity" class="section level3">
<h3><span class="header-section-number">10.1.2</span> Sparsity</h3>
<p>The data is sparse:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(<span class="kw">log10</span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span><span class="kw">abundances</span>(d)), <span class="dv">100</span>)</code></pre></div>
<p><img src="08-MAW-PVII_files/figure-html/pooled3-1.png" width="768" /></p>
</div>
<div id="rarity" class="section level3">
<h3><span class="header-section-number">10.1.3</span> Rarity</h3>
<p>Long tails of rare taxa:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(reshape2)
medians &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">abundances</span>(d),<span class="dv">1</span>,median)<span class="op">/</span><span class="fl">1e3</span>
A &lt;-<span class="st"> </span><span class="kw">melt</span>(<span class="kw">abundances</span>(d))
A<span class="op">$</span>Var1 &lt;-<span class="st"> </span><span class="kw">factor</span>(A<span class="op">$</span>Var1, <span class="dt">levels =</span> <span class="kw">rev</span>(<span class="kw">names</span>(<span class="kw">sort</span>(medians))))
p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(A, <span class="kw">aes</span>(<span class="dt">x =</span> Var1, <span class="dt">y =</span> value)) <span class="op">+</span>
<span class="st">        </span><span class="kw">geom_boxplot</span>() <span class="op">+</span>
<span class="st">    </span><span class="kw">labs</span>(<span class="dt">y =</span> <span class="st">&quot;Abundance (reads)&quot;</span>, <span class="dt">x =</span> <span class="st">&quot;Taxonomic Group&quot;</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw">scale_y_log10</span>()
    
<span class="kw">print</span>(p)</code></pre></div>
<p><img src="08-MAW-PVII_files/figure-html/tail-1.png" width="600px" /></p>
</div>
<div id="overdispersion" class="section level3">
<h3><span class="header-section-number">10.1.4</span> Overdispersion</h3>
<p>Variance exceeds the mean:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">means &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">abundances</span>(d),<span class="dv">1</span>,mean)
variances &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">abundances</span>(d),<span class="dv">1</span>,var)

<span class="co"># Calculate mean and variance over samples for each taxon</span>
<span class="kw">library</span>(reshape2)
<span class="kw">library</span>(dplyr)
df &lt;-<span class="st"> </span><span class="kw">melt</span>(<span class="kw">abundances</span>(d))
<span class="kw">names</span>(df) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Taxon&quot;</span>, <span class="st">&quot;Sample&quot;</span>, <span class="st">&quot;Reads&quot;</span>)
df &lt;-<span class="st"> </span>df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">group_by</span>(Taxon) <span class="op">%&gt;%</span>
<span class="st">             </span><span class="kw">summarise</span>(<span class="dt">mean =</span> <span class="kw">mean</span>(Reads),
                   <span class="dt">variance =</span> <span class="kw">var</span>(Reads))

<span class="co"># Illustrate overdispersion</span>
<span class="kw">library</span>(scales)
p &lt;-<span class="st"> </span><span class="kw">ggplot</span>(df, <span class="kw">aes</span>(<span class="dt">x =</span> mean, <span class="dt">y =</span> variance)) <span class="op">+</span>
<span class="st">       </span><span class="kw">geom_point</span>() <span class="op">+</span>
<span class="st">       </span><span class="kw">geom_abline</span>(<span class="kw">aes</span>(<span class="dt">intercept =</span> <span class="dv">0</span>, <span class="dt">slope =</span> <span class="dv">1</span>)) <span class="op">+</span>
<span class="st">       </span><span class="kw">scale_x_log10</span>(<span class="dt">labels =</span> scales<span class="op">::</span>scientific) <span class="op">+</span>
<span class="st">       </span><span class="kw">scale_y_log10</span>(<span class="dt">labels =</span> scales<span class="op">::</span>scientific) <span class="op">+</span>
<span class="st">       </span><span class="kw">labs</span>(<span class="dt">title =</span> <span class="st">&quot;Overdispersion (variance &gt; mean)&quot;</span>)
<span class="kw">print</span>(p)</code></pre></div>
<p><img src="08-MAW-PVII_files/figure-html/pooled_overdispersion-1.png" width="600px" /></p>
</div>
</div>
<div id="generalized-linear-models-a-brief-overview" class="section level2">
<h2><span class="header-section-number">10.2</span> Generalized linear models: a brief overview</h2>
<p>Let us briefly discuss the ideas underlying generalized linear models.</p>
<p>The Generalized linear model (GLM) allows a richer family of probability distributions to describe the data. Intuitively speaking, GLMs allow the modeling of nonlinear, nonsymmetric, and nongaussian associations. GLMs consist of three elements:</p>
<ul>
<li><p>A probability distribution for the data (from exponential family)</p></li>
<li><p>A linear predictor targeting the mean, or expectation: <span class="math inline">\(Xb\)</span></p></li>
<li><p>A link function g such that <span class="math inline">\(E(Y) = \mu = g^{-1}(Xb)\)</span>.</p></li>
</ul>
<p>Let us fit Poisson with (natural) log-link just to demonstrate how generalized linear models could be fitted in R. We fit the abundance (read counts) assuming that the data is Poisson distributed, and the logarithm of its mean, or expectation, is obtained with a linear model. For further examples in R, you can also check the <a href="https://www.statmethods.net/advstats/glm.html">statmethods website</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load again the example data</span>
d &lt;-<span class="st"> </span>dietswap
df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Abundance =</span> <span class="kw">abundances</span>(d)[taxa,],
                 <span class="dt">Group =</span> <span class="kw">meta</span>(d)<span class="op">$</span>nationality)

res &lt;-<span class="st"> </span><span class="kw">glm</span>(Abundance <span class="op">~</span><span class="st"> </span><span class="dv">1</span>, <span class="dt">data =</span> df, <span class="dt">family =</span> <span class="st">&quot;poisson&quot;</span>)</code></pre></div>
<p>Investigate the model output:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw">kable</span>(<span class="kw">summary</span>(res)<span class="op">$</span>coefficients, <span class="dt">digits =</span> <span class="dv">5</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Estimate</th>
<th align="right">Std. Error</th>
<th align="right">z value</th>
<th align="right">Pr(&gt;|z|)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>(Intercept)</td>
<td align="right">5.02355</td>
<td align="right">0.00544</td>
<td align="right">922.6545</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>Note the link between mean and estimated coefficient (<span class="math inline">\(\mu = e^{Xb}\)</span>):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">mean</span>(df<span class="op">$</span>Abundance)</code></pre></div>
<pre><code>## [1] 151.9505</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">exp</span>(<span class="kw">coef</span>(res))</code></pre></div>
<pre><code>## (Intercept) 
##    151.9505</code></pre>
</div>
<div id="deseq2-differential-abundance-testing-for-sequencing-data" class="section level2">
<h2><span class="header-section-number">10.3</span> DESeq2: differential abundance testing for sequencing data</h2>
<div id="fitting-deseq2" class="section level3">
<h3><span class="header-section-number">10.3.1</span> Fitting DESeq2</h3>
<p>[DESeq2 analysis]((<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8" class="uri">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8</a>) accommodates those particular assumptions about sequencing data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Start by converting phyloseq object to deseq2 format</span>
<span class="kw">library</span>(DESeq2)
d &lt;-<span class="st"> </span>dietswap <span class="co"># Phyloseq data</span>
ds2 &lt;-<span class="st"> </span><span class="kw">phyloseq_to_deseq2</span>(d, <span class="op">~</span><span class="st"> </span>group <span class="op">+</span><span class="st"> </span>nationality)

<span class="co"># Run DESeq2 analysis (all taxa at once!)</span>
dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(ds2)

<span class="co"># Investigate results</span>
deseq.results &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">results</span>(dds))
deseq.results<span class="op">$</span>taxon &lt;-<span class="st"> </span><span class="kw">rownames</span>(<span class="kw">results</span>(dds))

<span class="co"># Sort (arrange) by pvalue and effect size</span>
<span class="kw">library</span>(knitr)
deseq.results &lt;-<span class="st"> </span>deseq.results <span class="op">%&gt;%</span>
<span class="st">                   </span><span class="kw">arrange</span>(pvalue, log2FoldChange)

<span class="co"># Print the result table</span>
<span class="co"># Let us only show significant hits</span>
knitr<span class="op">::</span><span class="kw">kable</span>(deseq.results <span class="op">%&gt;%</span>
<span class="st">               </span><span class="kw">filter</span>(pvalue <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>log2FoldChange <span class="op">&gt;</span><span class="st"> </span><span class="fl">1.5</span>),
         <span class="dt">digits =</span> <span class="dv">5</span>)</code></pre></div>
<table>
<thead>
<tr class="header">
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">lfcSE</th>
<th align="right">stat</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
<th align="left">taxon</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="right">1870.08336</td>
<td align="right">1.89495</td>
<td align="right">0.12667</td>
<td align="right">14.95965</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Clostridium difficile et rel.</td>
</tr>
<tr class="even">
<td align="right">1015.44245</td>
<td align="right">2.12453</td>
<td align="right">0.18359</td>
<td align="right">11.57241</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Klebisiella pneumoniae et rel.</td>
</tr>
<tr class="odd">
<td align="right">3335.72902</td>
<td align="right">3.02549</td>
<td align="right">0.28380</td>
<td align="right">10.66044</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Mitsuokella multiacida et rel.</td>
</tr>
<tr class="even">
<td align="right">1519.25023</td>
<td align="right">1.60976</td>
<td align="right">0.18026</td>
<td align="right">8.93027</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Enterobacter aerogenes et rel.</td>
</tr>
<tr class="odd">
<td align="right">2848.86627</td>
<td align="right">1.77112</td>
<td align="right">0.22906</td>
<td align="right">7.73226</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Megasphaera elsdenii et rel.</td>
</tr>
<tr class="even">
<td align="right">321.24275</td>
<td align="right">2.86529</td>
<td align="right">0.37602</td>
<td align="right">7.61997</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Serratia</td>
</tr>
<tr class="odd">
<td align="right">19.97851</td>
<td align="right">2.24305</td>
<td align="right">0.31335</td>
<td align="right">7.15820</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Aquabacterium</td>
</tr>
<tr class="even">
<td align="right">173.00079</td>
<td align="right">1.80120</td>
<td align="right">0.26252</td>
<td align="right">6.86114</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">Haemophilus</td>
</tr>
</tbody>
</table>
</div>
<div id="comparison-between-deseq2-and-standard-models" class="section level3">
<h3><span class="header-section-number">10.3.2</span> Comparison between DESeq2 and standard models</h3>
<p>For comparison purposes, assess significances and effect sizes based on Wilcoxon test.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">test.taxa &lt;-<span class="st"> </span><span class="kw">taxa</span>(d)
pvalue.wilcoxon &lt;-<span class="st"> </span><span class="kw">c</span>()
foldchange &lt;-<span class="st"> </span><span class="kw">c</span>()
<span class="cf">for</span> (taxa <span class="cf">in</span> test.taxa) {
  <span class="co"># Create a new data frame for each taxonomic group</span>
  df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">Abundance =</span> <span class="kw">abundances</span>(d)[taxa,],
                   <span class="dt">Log10_Abundance =</span> <span class="kw">log10</span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span><span class="kw">abundances</span>(d)[taxa,]),
                   <span class="dt">Group =</span> <span class="kw">meta</span>(d)<span class="op">$</span>nationality)
  <span class="co"># Calculate pvalue and effect size (difference beween log means)       </span>
  pvalue.wilcoxon[[taxa]] &lt;-<span class="st"> </span><span class="kw">wilcox.test</span>(Abundance <span class="op">~</span><span class="st"> </span>Group, <span class="dt">data =</span> df)<span class="op">$</span>p.value
  foldchange[[taxa]] &lt;-<span class="st"> </span><span class="kw">coef</span>(<span class="kw">lm</span>(Log10_Abundance <span class="op">~</span><span class="st"> </span>Group, <span class="dt">data =</span> df))[[<span class="dv">2</span>]]
}
<span class="co"># Correct p-values for multiple testing</span>
pvalue.wilcoxon.adjusted &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(pvalue.wilcoxon)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">plot</span>(deseq.results<span class="op">$</span>padj, pvalue.wilcoxon.adjusted,
  <span class="dt">xlab =</span> <span class="st">&quot;DESeq2 adjusted p-value&quot;</span>,
  <span class="dt">ylab =</span> <span class="st">&quot;Wilcoxon adjusted p-value&quot;</span>,
  <span class="dt">main =</span> <span class="st">&quot;P-value comparison&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v =</span> <span class="fl">0.05</span>, <span class="dt">h =</span> <span class="fl">0.05</span>, <span class="dt">lty =</span> <span class="dv">2</span>)

<span class="kw">plot</span>(deseq.results<span class="op">$</span>log2FoldChange, foldchange, 
  <span class="dt">xlab =</span> <span class="st">&quot;DESeq2&quot;</span>,
  <span class="dt">ylab =</span> <span class="st">&quot;Linear model&quot;</span>,
  <span class="dt">main =</span> <span class="st">&quot;Effect size comparison&quot;</span>)
<span class="kw">abline</span>(<span class="dv">0</span>,<span class="dv">1</span>)</code></pre></div>
<p><img src="08-MAW-PVII_files/figure-html/pooled_pcomp-1.png" width="960" /></p>
<p>For systematic comparisons between various methods for differential abundance testing, see <a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-017-0237-y">this paper</a>.</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="linear-models-the-role-of-covariates.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="multivariate-comparisons-of-microbial-community-composition.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
